extensions/email/src/main/java/org/mule/extension/email/internal/commands/ReplyCommand.java

```diff
@@ -0,0 +1,83 @@
+ * LICENSE.txt file.
+ */
+package org.mule.extension.email.internal.commands;
+
+import static org.apache.commons.collections.CollectionUtils.isEmpty;
+import static org.mule.extension.email.internal.util.EmailConnectorUtils.getAttributesFromMessage;
+import static org.mule.extension.email.internal.util.EmailConnectorUtils.mapToEmailAttributes;
+import org.mule.extension.email.api.EmailAttachment;
+import org.mule.extension.email.api.EmailAttributes;
+import org.mule.extension.email.api.EmailContent;
+import org.mule.extension.email.internal.exception.EmailException;
+import org.mule.runtime.api.message.MuleMessage;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.mail.Session;
+
+/**
+ * Represents the reply operation.
+ *
+ * @since 4.0
+ */
+public final class ReplyCommand
+{
+
+    public static final String NO_EMAIL_FOUND = "Cannot perform the reply operation if no email is provided";
+    private final SendCommand sendCommand = new SendCommand();
+
+    /**
+     * Replies an email message. The message will be sent to the addresses
+     * associated to the replyTo attribute in the {@link EmailAttributes} of
+     * the incoming {@code muleMessage}.
+     * <p>
+     * If no email message is found in the incoming {@link MuleMessage} this operation will fail.
+     *
+     * @param session     the {@link Session} through which the message is going to be sent.
+     * @param muleMessage the incoming {@link MuleMessage} from which the email is going to getPropertiesInstance the content.
+     * @param content     the content of the reply message.
+     * @param subject     the subject of the email. is none is set then one will be created using the subject from the replying email.
+     * @param from        the person who sends the email.
+     * @param headers     a custom set of headers.
+     * @param replyToAll  if this reply should be sent to all recipients of this message, or only the sender of the received email.
+     */
+    public void reply(Session session,
+                      MuleMessage muleMessage,
+                      EmailContent content,
+                      String subject,
+                      String from,
+                      Map<String, String> headers,
+                      Boolean replyToAll)
+    {
+        EmailAttributes attributes = getAttributesFromMessage(muleMessage)
+                .orElseThrow(() -> new EmailException(NO_EMAIL_FOUND));
+
+        List<String> replyTo = attributes.getReplyToAddresses();
+        if (isEmpty(replyTo))
+        {
+            replyTo = attributes.getToAddresses();
+        }
+
+        if (subject == null)
+        {
+            subject = "Re: " + attributes.getSubject();
+        }
+
+        if (headers == null)
+        {
+            headers = new HashMap<>();
+        }
+
+        headers.put("In-Reply-To", Integer.toString(attributes.getId()));
```

Comment:
extract to a constnat and add a test which verifies that the responses have this header